### PR TITLE
Add missing api comments about ${force}

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

**Comment by gperciva** [comment]
This adds:
> If ${force} is 1, do not check whether decryption will exceed the estimated available memory or time.

to a few functions that should have it added.  (Unfortunately the branch name is a bit out of date, but we can't change that unless we close this PR and open a new one.)

Question: in https://github.com/Tarsnap/scrypt/pull/263, if somebody tried a `scryptenc` function with too-large explicit parameters, they received a `warn0` instead of an error.  However, we didn't document that in the function API.  Should I add that to this PR?

**Comment by gperciva** [comment]
I add the api comments about printing a warning instead of returning an error for `TOOBIG` or `TOOSLOW`.  If you don't like them, it's easy to remove that commit.

File: lib/scryptenc/scryptenc.c [hunk 1]
```diff
@@ -398,9 +398,12 @@ scryptdec_setup(const uint8_t header[96], uint8_t dk[64],
...
  *     params, verbose, force):
  * Encrypt ${inbuflen} bytes from ${inbuf}, writing the resulting
- * ${inbuflen} + 128 bytes to ${outbuf}.  The explicit parameters
- * within ${params} must be zero or must all be non-zero.  Return
- * the explicit parameters used via ${params}.
+ * ${inbuflen} + 128 bytes to ${outbuf}.  If ${force} is 1, do not check
+ * whether decryption will exceed the estimated available memory or time.
+ * The explicit parameters within ${params} must be zero or must all be
+ * non-zero.  If explicit parameters are used and the computation is estimated
+ * to exceed resource limits, print a warning to stderr instead of returning
```
Comment (gperciva) [inline lib/scryptenc/scryptenc.c:405]: That's not true; `warn0` can now send messages to syslog instead.  Maybe replace
```
print a warning to stderr
```
with
```
issue a warning
```
?

Comment (cperciva) [reply lib/scryptenc/scryptenc.c:405]: Let's drop the "to stderr" but keep "print a warning".

Comment (cperciva) [reply lib/scryptenc/scryptenc.c:405]: And in the description of `scryptenc_file` too.

**Comment by gperciva** [comment]
Fixed and ready for merge.

**Pull request merged**